lib/src/Base/Algo/openturns/CubaIntegration.hxx

```diff
@@ -0,0 +1,115 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Implement the Cuba method for multidimensional integration
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_CUBAINTEGRATION_HXX
+#define OPENTURNS_CUBAINTEGRATION_HXX
+
+#include "openturns/IntegrationAlgorithmImplementation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class CubaIntegration
+ */
+
+class OT_API CubaIntegration
+  : public IntegrationAlgorithmImplementation
+{
+
+  CLASSNAME
+
+public:
+
+  /** Default constructor without parameters */
+  CubaIntegration();
+
+  /** Parameter constructor */
+  explicit CubaIntegration(const String optRoutine);
+
+  /** Virtual copy constructor */
+  CubaIntegration * clone() const override;
+
+  /* Compute an approximation of \int_{I}f(x)dx, where I is the integration interval */
+  using IntegrationAlgorithmImplementation::integrate;
+  Point integrate(const Function & function,
+                  const Interval & interval) const override;
+
+  /** maximumRelativeError accessor */
+  Scalar getMaximumRelativeError() const;
+  void setMaximumRelativeError(const Scalar maximumRelativeError);
+
+  /** maximumAbsoluteError accessor */
+  Scalar getMaximumAbsoluteError() const;
+  void setMaximumAbsoluteError(const Scalar maximumAbsoluteError);
+
+  /** optRoutine accessor */
+  String getOptRoutine() const;
+  void setOptRoutine(const String optRoutine);
```

Comment:
nice progress!
There is also the method get/setAlgorithmName in other classes, we could use the same name here

Reply:
some optimization algorithms also have a GetAlgorithmNames static method that returns the available values,
then it can be used in the test to loop over algorithms